osu.Game.Rulesets.Catch/Edit/Blueprints/Components/EditablePath.cs

```diff
@@ -92,7 +92,7 @@ public void InitializeFromHitObject(JuiceStream hitObject)
             }));
         }
 
-        public void UpdateHitObjectFromPath(JuiceStream hitObject)
+        public virtual void UpdateHitObjectFromPath(JuiceStream hitObject)
```

Comment:
Reason for having this virtual and the actual logic only live in the selection subclass is that the logic breaks placement if put here. There are other protections in place already against placement allowing zero length:

https://github.com/ppy/osu/blob/e7070bd8120460562330957d24a7b99b41c6092a/osu.Game.Rulesets.Catch/Edit/Blueprints/JuiceStreamPlacementBlueprint.cs#L28